### ISSUE TITLE
FlashSize Value Cannot Be Changed

### ISSUE THREAD

**Issue by P0W3** [visualboyadvance-m/visualboyadvance-m#585]
I changed the value to 131072 in FlashSize from vbam.ini. But when I opened VisualBoyAdvance-M, the message stated that the value, 131072, is invalid.

![New Bitmap Image](https://user-images.githubusercontent.com/18279115/71563879-04dedf80-2a65-11ea-945e-323dd45e9ac2.png)

![MWV_0](https://user-images.githubusercontent.com/18279115/71563901-7fa7fa80-2a65-11ea-861c-13095086c0be.png)

**Comment by negativeExponent** [comment]
user error. you are not suppose to modify flash size from there. you force flashSize using the Options > Game Boy Advance > Game Overrides on a per-game-basis

**Comment by denisfa** [comment]
@P0W3 The option `flashSize` on `vbam.ini` is meant to be:

- flashSize=0 => 64KB 
- flashSize=1 => 128KB

If you want to modify it due to a game, then please do the change on
`Options` > `Game Boy Advance` > `Configure ...`

Go to the tab `Game Overrides`. You should be able to see and change what you want.

Is there a reason you want to do this? Like the game you want play not saving/playing properly?
If that is the case, then can you please send us the info about the game(`File` > `ROM Information`)?

Nevertheless, the warning message could be more descriptive. I will keep this open at the very least as an enhancement.

**Comment by negativeExponent** [comment]
```


Is there a reason you want to do this? Like the game you want play not saving/playing properly?
If that is the case, then can you please send us the info about the game(File > ROM Information)?
```
there is practically no reason to have two overrides! so i dont understand the idea to entertain the concept. in anycase if not using overrides, gba's findsavetype() will take care of finding the most compatible save format based on signature of the rom itself (similar to the other issue about gba pokemon red romhacks) which will then override optFlashSize anyways. im sorry to say but i don't think you guys know whats going on other that what you can do with the UI and just playing around with these labels...

**Comment by P0W3** [comment]
@denisfa Changing the flash size from 0 to 131072 is probably an old-fashioned method, based on what I did in vba-over.ini, but I am not using the game override settings.

When I used VBA-M (without editing the vbam.ini file), I changed the flash size from 64K to 128K (more specifically, 65536 to 131072). Plus, when I closed VBA-M and opened VBA-M, the flash size has been automatically switched back to 64K, and that was the buggy part.

The other part, when you mentioned about the flash size that can be changed from 0 to 1, is that I did change the flash size from 0 to 1 in the vbam.ini file. And when I opened VBA-M, the flash size keeps 128K, based on what I did in the vbam.ini file.

I did not play a game, while I was changing the flash size settings.

As for the warning message, it should be changed from this, "Invalid value ______ for option preferences/flashSize; valid values are 0 - 0.", to this, "Invalid value ______ for option preferences/flashSize; valid values are 0 - 1.".
